src/store.rs

```diff
@@ -38,28 +36,32 @@ impl Store {
         self.path.join(component_dirname(component_name, version))
     }
 
+    pub(crate) fn install_component(&self, cfg: &DownloadCfg) -> Result<()> {
+        let component_dir = self.component_dir_path(&cfg.name, &cfg.version);
+
+        ensure_dir_exists(&component_dir)?;
+        download_file_and_unpack(cfg, &component_dir)?;
+        // We ensure that component_dir exists above, so its parent must exist here.
+        unpack_bins(&component_dir, &component_dir)?;
+
+        Ok(())
+    }
+
     // This function installs a component into a directory within '/.fuelup/store'.
     // The directory is named '<component_name>-<version>', eg. 'fuel-core-0.15.1'.
-    pub(crate) fn install_component(
+    pub(crate) fn install_toolchain_component(
```

Comment:
Maybe it's better that the `Store` doesn't know anything about `toolchain`s, and just handles `components`? We could leave it up to the `toolchain` to create its own links. This might require the `install_component` method returning some info about what was installed (e.g. the resulting path). Will leave it up to you if you want to make this change though!